reana_job_controller/app.py

```diff
@@ -55,212 +59,195 @@ def filter_jobs(job_db):
-       or not ('experiment') in request.json\
-       or not ('docker-img' in request.json):
-        print(request.json)
-        abort(400)
-
-    cmd = request.json['cmd'] if 'cmd' in request.json else None
-    env_vars = (request.json['env-vars']
-                if 'env-vars' in request.json else {})
-
-    if request.json.get('cvmfs_mounts'):
-        cvmfs_repos = request.json.get('cvmfs_mounts')
-    else:
-        cvmfs_repos = []
+    # Validate and deserialize input
+    job_request, errors = job_request_schema.load(json_data)
 
-    job_id = str(uuid.uuid4())
+    if errors:
+        return jsonify(errors), 400
 
-    job_obj = instantiate_job(job_id,
-                              request.json['docker-img'],
-                              cmd,
-                              cvmfs_repos,
-                              env_vars,
-                              request.json['experiment'],
-                              shared_file_system=True)
+    job_obj = instantiate_job(job_request['job_id'],
+                              job_request['docker_img'],
+                              job_request['cmd'],
+                              job_request['cvmfs_mounts'],
+                              job_request['env_vars'],
+                              job_request['experiment'],
+                              job_request['shared_file_system'])
 
     if job_obj:
-        job = copy.deepcopy(request.json)
-        job['job-id'] = job_id
+        job = copy.deepcopy(job_request)
         job['status'] = 'started'
         job['restart_count'] = 0
         job['max_restart_count'] = 3
         job['obj'] = job_obj
         job['deleted'] = False
-        JOB_DB[job_id] = job
-        return jsonify({'job-id': job_id}), 201
+        JOB_DB[job['job_id']] = job
+        return jsonify({'job_id': job['job_id']}), 201
     else:
         return jsonify({'job': 'Could not be allocated'}), 500
 
 
 @app.route('/jobs/<job_id>', methods=['GET'])
-def get_job(job_id):
-    """Get a job.
-
-    FIXME --> probably this endpoint should be merged with `get_jobs()`
-
-    .. http:get:: /jobs/<job_id>
-
-        Returns a JSON list with all the jobs.
-
-        **Request**:
-
-        .. sourcecode:: http
-
-            GET /jobs/cdcf48b1-c2f3-4693-8230-b066e088c6ac HTTP/1.1
-            Content-Type: application/json
-            Host: localhost:5000
-
-        :reqheader Content-Type: application/json
-
-        **Responses**:
-
-        .. sourcecode:: http
-
-            HTTP/1.0 200 OK
-            Content-Length: 80
-            Content-Type: application/json
-
-            {
+def get_job(job_id):  # noqa
+    """Get a Job.
+
+    ---
+    get:
+      description: Get a Job by its id
+      produces:
+       - application/json
+      parameters:
+       - name: job_id
+         in: path
+         description: ID of the Job
+         required: true
+         type: string
+      responses:
+        200:
+          description: The Job.
```

Comment:
Ditto for description of responses:

- 200 Request succeeded. The response contains details about the given job ID.
- 404 Request failed. The given job ID does not seem to exist.